sandbox_dns_unix.go

```diff
@@ -184,6 +187,13 @@ func (sb *sandbox) setupDNS() error {
 			return err
 		}
 	} else {
+		// If embedded DNS server is enabled later for this contianer ipv4 loopback nameservers if
+		// present should be used. retain it before filtering out resolv.conf contents.
```

Comment:
Lets say on docker run the container is connected to the default bridge and the host's resolv.conf has only 127.0.0.1 as the resolver. In the filter logic this would be removed and the default resolvers will be added. Later when the container is connected to a user defined network, we enable the embedded DNS server. But at that time we can't tell if the default servers were retrieved from the host's resolv.conf or we added it in the filtering logic. If its the latter then we have to use the loopback as external server and remove the default servers. So what I am doing now is to construct `sb.extDNS` before we do the filtering and use it later when starting the resolver.